src/components/index/indexPresenter.tsx

```diff
@@ -22,18 +23,26 @@ const IndexPresenter: React.FC = () => {
 
   const [data, setData] = useRecoilState(algorithmState);
   const [isHasNext, setIsHasNext] = useState(true);
-  let hasNext = true;
+  let hasNext: boolean | undefined = true;
   let cursor2: number | undefined;
 
   const getPostList = () => {
-    let posts: algorithm[];
-    Post.getPost(isAdmin, cursor2, algorithmFilter).then((res) => {
-      posts = res.data.posts;
-      cursor2 = res.data.cursor;
-      hasNext = res.data.hasNext;
-      setData([data.concat(posts)][0]);
-      setIsHasNext(res.data.hasNext);
-    });
+    let posts: algorithm[] | undefined;
+    Post.getPost(isAdmin, cursor2, algorithmFilter).then(
+      (res: AxiosResponse<getPostRes> | void) => {
+        posts = res?.data?.posts;
+        cursor2 = res?.data.cursor;
+        hasNext = res?.data.hasNext;
+        setData(
```

Comment:
```suggestion
        setData([...data, ...posts || [{ number: 0, createdAt: 0, id: "", status: "" }]][0]);
```
느낌의 코드를 작성하는 것 같은데, 위의 문법은 어떨까요?

Reply:
이유가 무엇이실까요?

Reply:
노션에서 마저 이야기를 나누도록 해요 :)